packages/plugins/src/hardhat/simulate.ts

```diff
@@ -25,24 +25,41 @@ import {
   removeRoles,
   fetchNetworkConfigFromDeploymentConfig,
   NetworkConfig,
-  callWithTimeout,
   fetchExecutionTransactionReceipts,
   convertEthersTransactionReceipt,
   InvariantError,
   getContractAddressesFromNetworkConfig,
   sphinxCoreExecute,
   convertEthersTransactionResponse,
   SphinxTransactionResponse,
+  sphinxCoreUtils,
 } from '@sphinx-labs/core'
 import { ethers } from 'ethers'
 import { HardhatEthersProvider } from '@nomicfoundation/hardhat-ethers/internal/hardhat-ethers-provider'
 import pLimit from 'p-limit'
 
+import {
+  HardhatResetNotAllowedErrorMessage,
+  getRpcRequestStalledErrorMessage,
+} from '../foundry/error-messages'
+
 export type SimulationTransactions = Array<{
   receipt: SphinxTransactionReceipt
   response: SphinxTransactionResponse
 }>
 
+/**
+ * @property maxAttempts - The maximum number of attempts that Sphinx will make for a single network
+ * request before throwing an error.
+ * @property timeout - The maximum number of time that Sphinx will wait for a single RPC request
+ * before timing out. An RPC request can stall if the RPC provider has degraded service, which would
+ * cause the simulation to stall indefinitely if we don't time out.
+ */
+export const simulationConstants = {
+  maxAttempts: 7,
+  timeout: 90_000,
```

Comment:
Nit: Since this timeout is for every single call, 90 seconds is a very long timeout period. I think something as low as 10 seconds would be acceptable.

Reply:
I reduced it to 60s, but I don't think we should set it much lower than that. Added a bit more context on this to the `simulationConstants` variable:
```
* We set this value to be
 * relatively high because the execution process may submit very large transactions (specifically
 * transactions with `EXECUTE` Merkle leaves), which can cause the RPC request to be slow.
```